### ISSUE TITLE
Having FCC Eslint Rules Be Importable

### ISSUE THREAD

**Issue by jnmorse** [freeCodeCamp/testable-projects-fcc#232]
In Issue [120](https://github.com/freeCodeCamp/testable-projects-fcc/issues/120) of testable-projects-fcc, I was discussing having an importable eslint config. This would require a separate repository to be set up for it.

I created a repository here https://github.com/jnmorse/eslint-config-fcc that does include the fcc rules and can be installed as a dev package to serve as an example. This package is using a more up to date release of eslint 4.7.2.

Think it would be best if the package that is used was published to npm so it could be installed like a normal package.

At @tchaffee suggestion, I am opening the issue and will post in the main repo as well for discussion.

The benefits to having eslint-config package are that it can be maintained and used for multiple projects, and campers could include it as a dependency for their own projects. The downside is that would be external and all changes will have to happen there, aside from any project specific rules.

**Comment by tchaffee** [comment]
@jnmorse Thanks for opening a separate issue. I think we should break this down further into two separate issues? As I understand it, there are two goals?

1) Create a shareable eslint config that could be shared between all freeCodeCamp projects so any changes are reflected in all projects. Campers could also include it in their projects if they wanted.

2) Once that is finished we could discuss (most likely Quincy and other top devs) about any additional functionality like eslint-find-rules.

Does that make sense as a plan? Thanks!

**Comment by jnmorse** [comment]
@tchaffee Okay, I edited the issues to remove the references to what tooling I used in the repo and just left the link to the repo I made for example purposes.

**Comment by tchaffee** [comment]
@jnmorse Thanks. What would be the next steps to implement this? Does FCC need to publish a package to npm, or could it be done with a new FCC repo on github first?

**Comment by jnmorse** [comment]
@tchaffee It could be done via a repo to start with. Though publishing it npm would make it much easier to work with.

To install it from a repo as you'd do `npm i -D user/eslint-config-fcc` where "user" is the account its on, for instance `freeCodeCamp/eslint-config-fcc`. see [npm-install documentation](https://docs.npmjs.com/cli/install) for more info

**Comment by tchaffee** [comment]
@jnmorse That sounds good. Let's start with that and see what Quincy and the rest of the team think about it. IMO, it's a big improvement over maintaining separate files for each project. I'm going to continue the discussion over at the issue on the main project.
